source/dem/particle_wall_nonlinear_force.cc

```diff
@@ -223,6 +227,190 @@ ParticleWallNonLinearForce<dim>::calculate_particle_wall_contact_force(
+
+                      const Point<3> &projection_point =
+                        projection_points[particle_counter];
+
+                      Point<3> particle_location_3d;
+
+                      if constexpr (dim == 3)
+                        particle_location_3d = particle->get_location();
+
+                      if constexpr (dim == 2)
+                        particle_location_3d =
+                          point_nd_to_3d(particle->get_location());
+
+                      const double particle_triangle_distance =
+                        particle_location_3d.distance(projection_point);
+
+                      // Find normal overlap
+                      double normal_overlap =
+                        ((particle_properties[DEM::PropertiesIndex::dp]) *
+                         0.5) -
+                        particle_triangle_distance;
+
+                      if (normal_overlap > 0)
+                        {
+                          contact_info.normal_overlap = normal_overlap;
+                          // Update normal vector
+                          contact_info.normal_vector =
+                            normal_vectors[particle_counter];
+                          contact_info.point_on_boundary = projection_point;
+
+                          contact_info.boundary_id = solid_counter;
+
+                          this
+                            ->update_particle_floating_wall_contact_information(
+                              contact_info,
+                              particle_properties,
+                              dt,
+                              translational_velocity,
+                              rotational_velocity,
+                              center_of_rotation.distance(
+                                particle_location_3d));
+
+                          // This tuple (forces and torques) contains four
+                          // elements which are: 1, normal force, 2, tangential
+                          // force, 3, tangential torque and 4, rolling
+                          // resistance torque, respectively
+                          std::tuple<Tensor<1, 3>,
+                                     Tensor<1, 3>,
+                                     Tensor<1, 3>,
+                                     Tensor<1, 3>>
+                            forces_and_torques =
+                              this
+                                ->calculate_nonlinear_contact_force_and_torque(
+                                  contact_info, particle_properties);
+
+                          // Getting particle's torque and force
+#if (DEAL_II_VERSION_MAJOR < 10 && DEAL_II_VERSION_MINOR < 4)
+                          types::particle_index particle_id =
+                            particle->get_id();
+#else
+                          types::particle_index particle_id =
+                            particle->get_local_index();
+#endif
```

Comment:
```suggestion
                          types::particle_index particle_id =
                            particle->get_local_index();
```

We don't support 9.3 anymore.

Reply:
Applied.